File: checks/check-balance.pl
#!/usr/bin/perl
use Term::ANSIColor qw(:constants);

print "Checking balance... ";

open (ACCOUNT, "goal account dump -a $ENV{addr}|");
local $/;
$account = <ACCOUNT>;
$account =~ /"algo": (\d+)/;
$balance = $1;
if ($balance >= 550000000)
{ 
  print GREEN "OK - " . $balance / 1000000 .  " voi is enough\n" . RESET;
} 
else 
{
  print RED $balance / 1000000 . " is NOT enough\n" . RESET;
}

File: checks/check-balance.py
#!/usr/bin/python3
import re
import os
from helper import *

lowVOI = 100 
suggestedVOI = 550

printHEADER("Checking Balance")

goal = runCommand(["goal", "account", "dump", "-a", os.environ["addr"]])
goalcmd = re.search(r"\"algo\"\:\s+(\d+)",goal)
balance = int(goalcmd.group(1))

if (balance >= (suggestedVOI * 1000000)):
  printOK(str(round(balance / 1000000,2)) + " VOI is enough")
elif (balance <= (lowVOI * 1000000)): 
  printERROR(str(round(balance / 1000000,2)) + " VOI is a _lot_ less than optimal - add via the Discord faucet until you have more VOI until you have " + str(suggestedVOI))
else:
  printWARNING(str(round(balance / 1000000,2)) + " VOI is less than optimal (" + str(suggestedVOI) +" or more)")
